contracts/ens/cid_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package ens
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	eipSpecHash = "e3010170122029f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f"
+	eipHash     = "29f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f"
+
+	dag_pb   = 0x70
+	sha2_256 = 0x12
+)
+
+// Tests for the decoding of the example ENS
+func TestEIPSpecCidDecode(t *testing.T) {
```

Comment:
Nitpick but tests like this are much more readable when they are in table-driven format, and you have to `input` and `wantXXX` (expectations) next to each other. Currently you have to look for `const`s outside the test and you have to read carefully the test to see what's compared where - otherwise you know that all `wantXXX` expectations are checked.

Reply:
cool. i copied the consts into the individual tests for readability sake. this specific test cannot go into the table format of the test below since it unit tests a different function. is this what you've meant?